aiida/orm/nodes/data/structure.py

```diff
@@ -896,16 +896,23 @@ def build_kind_name(species_and_occu):
         self.pbc = [True, True, True]
         self.clear_kinds()
 
+        pymatgen_version = tuple(map(int, get_pymatgen_version().split('.')))
         for site in struct.sites:
 
+            # site.species property first introduced in pymatgen version 2019.3.13
+            if pymatgen_version < (2019, 3, 13):
```

Comment:
I think here it'd be better to just do plain string comparison - something like ``"2019.3.13b2"`` would also be a legal version string (although pymatgen doesn't seem to use it). That would break the ``int`` conversion above.

Sorry for the nit-pick :grimacing:

Reply:
Better still I would say is to use something like:
```
from distutils.version import StrictVersion
current_version = StrictVersion(get_pymatgen_version())
required_version = StrictVersion('2019.3.13')
if current_version < required_version:
    species_and_occu = site.species_and_occu
```

Reply:
Nice :+1:

Btw, do we want to bump the ``pymatgen`` version in ``setup.json``? It's currently at ``"pymatgen<=2018.12.12"``. Not sure what the differences are though, tbh.

This issue came up because some dependency changed the ``pymatgen`` version.

Reply:
We can't unfortunately because they dropped python 2 support after that.

Reply:
Soon... :sunglasses: